repository-type/repository-list-by-type/demo.js

```diff
@@ -0,0 +1,84 @@
+visualize({
+    auth: {
+        name: "joeuser",
+        password: "joeuser",
+        organization: "organization_1"
+    }
+}, function(v) {
+
+    var resourceTypes = ['folder',
+        'dataType',
+        'jdbcDataSource',
+        'awsDataSource',
+        'jndiJdbcDataSource',
+        'virtualDataSource',
+        'customDataSource',
+        'beanDataSource',
+        'xmlaConnection',
+        'listOfValues',
+        'file',
+        'reportOptions',
+        'dashboard',
+        'adhocDataView',
+        'query',
+        'olapUnit',
+        'reportUnit',
+        'domainTopic',
+        'semanticLayerDataSource',
+        'secureMondrianConnection',
+        'mondrianXmlaDefinition',
+        'mondrianConnection',
+        'inputControl'
+    ];
+
+    var sortBySelectElement = document.getElementById('sortBy');
+
+    var resourceTypeElement = document.getElementById('resourceType');
+
+    resourceTypes.forEach(function(i) {
+        let option = document.createElement('option');
+        option.text = i;
+        option.label = i;
+        resourceTypeElement.add(option)
+    });
+
+    var search = v.resourcesSearch({
+        folderUri: "/public",
+        types: ["reportUnit"],
+        success: renderResults,
+        error: displayError
+    });
+
+    document
+        .getElementById('resourceType')
+        .addEventListener("change", function() {
+            let res = document.getElementById('resourceType').value;
```

Comment:
Please rename `res` variable name to `resourceType` so it will be more clear for the reader of the code.